runtime/t0rn-parachain/src/circuit_config.rs

```diff
@@ -1,19 +1,102 @@
-use crate::{AccountId, AccountManager, Balances, Call, Event, Portal, Runtime, Timestamp, XDNS};
+use crate::{
+    AccountId, AccountManager, Balances, BlockNumber, Call, Circuit, Clock, Event, Portal, Runtime,
+    Timestamp, XDNS,
+};
 
-use frame_support::{parameter_types, traits::ConstU32, PalletId};
+use frame_support::{parameter_types, traits::ConstU32, weights::Weight, PalletId};
 use pallet_grandpa_finality_verifier::bridges::runtime as bp_runtime;
 use sp_core::H256;
-use sp_runtime::traits::{BlakeTwo256, Convert};
-
+use sp_runtime::{
+    traits::{BlakeTwo256, Convert, One},
+    Perbill,
+};
 impl t3rn_primitives::EscrowTrait<Runtime> for Runtime {
     type Currency = Balances;
     type Time = Timestamp;
 }
 
+pub struct GlobalOnInitQueues;
+
+impl pallet_clock::traits::OnHookQueues<Runtime> for GlobalOnInitQueues {
+    fn process(n: BlockNumber, on_init_weight_limit: Weight) -> Weight {
+        let mut weights_consumed = vec![];
+        // Iterate over all pre-init hooks implemented by pallets and return aggregated weight
+        weights_consumed.push(Circuit::process_signal_queue(
+            n,
+            BlockNumber::one(),
+            Perbill::from_percent(15) * on_init_weight_limit,
+        ));
+        log::debug!(
+            "Circuit::process_signal_queue consumed: {:?}",
+            weights_consumed
+                .last()
+                .expect("Circuit::process_signal_queue consumed weight")
+        );
+        weights_consumed.push(Circuit::process_xtx_tick_queue(
```

Comment:
consider separate trait for all "process...(n: BlockNumber, interval: BlockNumber, limit: Weight)